merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/driver/Scheduler.java

```diff
@@ -11,9 +11,11 @@
 
   <Event> void emit(Event event, Query<? super Event, ?> query);
 
-  String spawn(Task task);
-  String spawn(String type, Map<String, SerializedValue> arguments);
+  TaskIdentifier spawn(Task task);
+  TaskIdentifier spawn(String type, Map<String, SerializedValue> arguments);
 
-  String defer(Duration delay, Task task);
-  String defer(Duration delay, String type, Map<String, SerializedValue> arguments);
+  TaskIdentifier defer(Duration delay, Task task);
+  TaskIdentifier defer(Duration delay, String type, Map<String, SerializedValue> arguments);
+
+  interface TaskIdentifier {}
```

Comment:
I called this `TaskIdentifier` so that I wouldn't have to use fully qualified names in the `SimulationEngine`. If anyone thinks `TaskId` (or any other name) would be better, happy to change it.

Reply:
Not to make more work for you but I'm on team `TaskId` :) This would follow the `ActivityInstanceId` and `PlanId` convention we have.

Reply:
Have there been any discussions yet on removing `waitFor`? If we did that, we could remove the need to return a task identifier in the `Scheduler` interface altogether.

Reply:
Meanwhile, I'm also on team `TaskId` -- qualified name collisions are the name of the game when it comes to complex component boundaries like this. You could call the engine's version an `EngineTaskId` or `ConcreteTaskId` instead, if you prefer.

Reply:
> Have there been any discussions yet on removing `waitFor`? If we did that, we could remove the need to return a task identifier in the `Scheduler` interface altogether.

Not yet - punting on that until "Step 2: (AERIE-1630) Make this value available to the caller."

Sounds good - thanks, team TaskId 😉

Reply:
I chatted with Pat a bit this evening, and threw together a branch for removing `waitFor` at `feature/remove-waitFor`. This would get rid of the need to return task identifiers at all, which is *lovely*.

We just need to make sure this won't adversely impact customers; Pat has kindly offered to do a cursory investigation on this front.